packages/grid/_modules_/grid/components/column-header-item.tsx

```diff
@@ -1,120 +1,137 @@
 import { OptionsContext } from './options-context';
 import { CursorCoordinates } from '../hooks/features/useColumnReorder';
 
 interface ColumnHeaderItemProps {
-  column: ColDef;
   colIndex: number;
-  onResizeColumn?: (c: any) => void;
-  onColumnDragStart?: (col: ColDef, currentTarget: HTMLElement) => void;
+  column: ColDef;
   onColumnDragEnter?: (event: Event) => void;
   onColumnDragOver?: (col: ColDef, coordinates: CursorCoordinates) => void;
+  onColumnDragStart?: (col: ColDef, currentTarget: HTMLElement) => void;
+  separatorProps: React.HTMLAttributes<HTMLDivElement>;
 }
 
-export const ColumnHeaderItem = React.memo(
-  ({
-    column,
+export const ColumnHeaderItem = React.memo((props: ColumnHeaderItemProps) => {
+  const {
     colIndex,
-    onResizeColumn,
-    onColumnDragStart,
+    column,
     onColumnDragEnter,
     onColumnDragOver,
-  }: ColumnHeaderItemProps) => {
-    const api = React.useContext(ApiContext);
-    const { showColumnRightBorder, disableColumnResize, disableColumnReorder } = React.useContext(
-      OptionsContext,
-    );
+    onColumnDragStart,
+    separatorProps,
+  } = props;
+  const apiRef = React.useContext(ApiContext) as ApiRef;
+  const { showColumnRightBorder, disableColumnResize, disableColumnReorder } = React.useContext(
+    OptionsContext,
+  );
+
+  const [resizing, setResizing] = React.useState(false);
+  const handleResizeStart = React.useCallback(
+    (params) => {
+      if (column.field === params.field) {
+        setResizing(true);
+      }
+    },
+    [column.field],
+  );
+  const handleResizeStop = React.useCallback(() => {
+    setResizing(false);
+  }, []);
+  useApiEventHandler(apiRef, COL_RESIZE_START, handleResizeStart);
+  useApiEventHandler(apiRef, COL_RESIZE_STOP, handleResizeStop);
 
-    let headerComponent: React.ReactElement | null = null;
-    if (column.renderHeader) {
-      headerComponent = column.renderHeader({
-        api: api!.current!,
-        colDef: column,
-        colIndex,
-        field: column.field,
-      });
-    }
+  let headerComponent: React.ReactElement | null = null;
+  if (column.renderHeader) {
+    headerComponent = column.renderHeader({
+      api: apiRef.current!,
+      colDef: column,
+      colIndex,
+      field: column.field,
+    });
+  }
 
-    const handleResize = onResizeColumn && (() => onResizeColumn(column));
-    const dragConfig = {
-      draggable:
-        !disableColumnReorder && !!onColumnDragStart && !!onColumnDragEnter && !!onColumnDragOver,
-      onDragStart: onColumnDragStart && ((event) => onColumnDragStart(column, event.currentTarget)),
-      onDragEnter: onColumnDragEnter && ((event) => onColumnDragEnter(event)),
-      onDragOver:
-        onColumnDragOver &&
-        ((event) => {
-          onColumnDragOver(column, {
-            x: event.clientX,
-            y: event.clientY,
-          });
-        }),
-    };
-    const width = column.width!;
+  const dragConfig = {
```

Comment:
usememo? avoid to recreate the functions and rerender the subtree...

Reply:
Mind that if you review with:

<img width="221" alt="Capture d’écran 2020-10-19 à 15 42 29" src="https://user-images.githubusercontent.com/3165635/96458885-be3fb900-1221-11eb-9c7a-1f81259c17f6.png">

You will see that this line is not linked to the resizing change.

Reply:
But the value is spread, so no performance issue here

Reply:
the value is spread but when you do `(event)=> something(event)` it creates a new function on each render and so shallow equal will never work and the sub tree will  always rerender

Reply:
It's applied on a div, so no shallow win to gain